### ISSUE TITLE
[BUG] Multi-language documents produce unstable language metadata

### ISSUE THREAD

**Issue by hansmi** [paperless-ngx/paperless-ngx#5607]
### Description

Paperless-ngx uses the [`langdetect` Python module](https://pypi.org/project/langdetect/) to determine the human language a document is written in. The language is detected every time the document metadata is retrieved via the API (`/api/documents/<id>/metadata/`).

https://github.com/paperless-ngx/paperless-ngx/blob/d55900b8770ff5f9a85523aa3e8638f98a30cc2b/src/documents/views.py#L422-L427

The detection algorithm used by `langdetect` is non-deterministic, relying on a random number generator. From the documentation ([source](https://github.com/Mimino666/langdetect/blob/a1598f1afcbfe9a758cfd06bd688fbc5780177b2/README.md?plain=1#L48-L57)):

> Language detection algorithm is non-deterministic, which means that if you try to run it on a text which is either too short or too ambiguous, you might get different results everytime you run it.

For multi-language documents requesting the metadata multiple times will randomly return varying languages.

I see the following options to make the metadata stable:

1. [+] Language detection isn't the fastest to begin with. Storing a two-character string among the rest of the metadata isn't the end of the world. As such the language could be detected once after OCR and stored.
1. [-] Instantiate a custom `langdetect.DetectorFactory` with its seed set to a fixed value (zero?). I'd avoid modifying the global seed as suggested by the upstream documentation in case the module is used elsewhere in the process.
1. [-] `langdetect.detect_langs` returns the probabilities. The language could be returned as `none` or similar if it's undecided beyond a certain cutoff probability.

Opinions?

### Steps to reproduce

I put the text extracted from the [Swiss Federal Tax Administration's salary certificate in German, French and Italian (`dbst-form-11lohna-rechts-dfi-de.pdf`)](https://www.estv.admin.ch/estv/de/home/direkte-bundessteuer/lohnausweis.html) into `/tmp/content.txt` and ran a small Python snippet:

```python
import langdetect, collections

content = open("/tmp/content.txt").read();
stats = collections.defaultdict(int)

for _ in range(1000):
  stats[langdetect.detect(content)] += 1

print(sorted(stats.items()))
```

The output will look something like the following:

```python
[('de', 24), ('fr', 47), ('it', 929)]
```

For the API the expected result would be a single language. Which doesn't really matter in this instance as long as it's stable.

### Webserver logs

```bash
n/a
```

### Paperless-ngx version

2.4.2

### Host OS

Debian Linux

### Installation method

Docker - official image

### Please confirm the following

- [X] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [X] I have already searched for relevant existing issues and discussions before opening this report.
- [X] I have updated the title field above with a concise description.

**Comment by stumpylog** [comment]
This seems like a feature request, not a bug.

**Comment by hansmi** [comment]
It's somewhere between a feature request and a bug indeed. The API not always returning the same data for an otherwise untouched document is a bug, the way to fix it is the feature request.

**Comment by stumpylog** [comment]
>  The API not always returning the same data for an otherwise untouched document is a bug

I would disagree.  There's no guarantees about what the metadata will return.  It's format isn't even documented.  In particular the language is only used for setting RTL display or not (see da38efebdf46c2a06b86a7615406522f93e66b0c).

I would make a feature request about wanting it to be a stable return.  And including a why it would matter.
